plugins/modules/net_tools/dnsimple.py

```diff
@@ -201,135 +393,121 @@ def main():
     priority = module.params.get('priority')
     state = module.params.get('state')
     is_solo = module.params.get('solo')
+    sandbox = module.params.get('sandbox')
 
-    if account_email and account_api_token:
-        client = DNSimple(email=account_email, api_token=account_api_token)
-    elif os.environ.get('DNSIMPLE_EMAIL') and os.environ.get('DNSIMPLE_API_TOKEN'):
-        client = DNSimple(email=os.environ.get('DNSIMPLE_EMAIL'), api_token=os.environ.get('DNSIMPLE_API_TOKEN'))
-    else:
-        client = DNSimple()
+    DNSIMPLE_MAJOR_VERSION = LooseVersion(dnsimple_version).version[0]
 
     try:
+        if DNSIMPLE_MAJOR_VERSION > 1:
+            ds = DNSimpleV2(account_email, account_api_token, sandbox, module)
+        else:
+            ds = DNSimpleV1(account_email, account_api_token, sandbox, module)
         # Let's figure out what operation we want to do
-
         # No domain, return a list
         if not domain:
-            domains = client.domains()
-            module.exit_json(changed=False, result=[d['domain'] for d in domains])
+            all_domains = ds.get_all_domains()
+            module.exit_json(changed=False, result=all_domains)
 
         # Domain & No record
         if domain and record is None and not record_ids:
```

Comment:
Again, sorry, I see it was a previously existing code but, at this point in code, `if domain` is redundant, because if domain was not set, the module would have exited (in line 409).

Reply:
I see what you mean. Same applies to `if domain and record is not None` and `if domain and record_ids` I guess. I will take a look at it

Reply:
fixed in https://github.com/ansible-collections/community.general/pull/2946/commits/343528fe3ff846fc308862f76fc9d2bf842e5918